game-core/src/main/java/games/strategy/engine/lobby/moderator/toolbox/tabs/moderators/ModeratorsTab.java

```diff
@@ -0,0 +1,115 @@
+import org.triplea.java.Interruptibles;
+import org.triplea.swing.ButtonColumn;
+import org.triplea.swing.DocumentListenerBuilder;
+import org.triplea.swing.JButtonBuilder;
+import org.triplea.swing.JPanelBuilder;
+import org.triplea.swing.JTableBuilder;
+import org.triplea.swing.JTextFieldBuilder;
+import org.triplea.swing.SwingComponents;
+
+
+/**
+ * Tab with a table showing list of moderators. Super-mods will see additional controls.
+ */
+public final class ModeratorsTab implements Supplier<Component> {
+  private static final int USERNAME_MIN_LENGTH = 4;
+  private final JFrame parentFrame;
+  private final ModeratorsTabModel moderatorsTabModel;
+  private final ModeratorsTabActions moderatorsTabActions;
+  private volatile boolean checkUserNameIsInFlight = false;
+
+  public ModeratorsTab(
+      final JFrame parentFrame, final ToolboxModeratorManagementClient toolboxModeratorManagementClient) {
+    this.parentFrame = parentFrame;
+    this.moderatorsTabModel = new ModeratorsTabModel(toolboxModeratorManagementClient);
+    this.moderatorsTabActions = new ModeratorsTabActions(parentFrame, moderatorsTabModel);
+  }
+
+  @Override
+  public Component get() {
+    final JTable dataTable = buildTable();
+
+    final JPanel panel = JPanelBuilder.builder()
+        .border(10)
+        .addNorth(JButtonBuilder
+            .builder()
+            .title("Refresh")
+            .actionListener(() -> moderatorsTabActions.refreshTableData(dataTable))
+            .build())
+        .addCenter(SwingComponents.newJScrollPane(dataTable))
+        .build();
+
+    if (moderatorsTabModel.isSuperMod()) {
+      panel.add(buildAddModeratorPanel(dataTable), BorderLayout.SOUTH);
+    }
+    return panel;
+  }
+
+  private JTable buildTable() {
+    final JTable table = JTableBuilder.builder()
+        .columnNames(moderatorsTabModel.fetchTableHeaders())
+        .tableData(moderatorsTabModel.fetchTableData())
+        .build();
+    if (moderatorsTabModel.isSuperMod()) {
+      ButtonColumn.attachButtonColumn(table, 2, moderatorsTabActions.generateApiKeyAction());
+      ButtonColumn.attachButtonColumn(table, 3, moderatorsTabActions.removeModAction(parentFrame));
+      ButtonColumn.attachButtonColumn(table, 4, moderatorsTabActions.addSuperModAction(parentFrame));
+    }
+    return table;
+  }
+
+  private Component buildAddModeratorPanel(final JTable table) {
+    final JTextField addField = JTextFieldBuilder.builder()
+        .columns(7)
+        .maxLength(30)
+        .toolTip("Username to add to moderators")
+        .build();
+
+    final JButton addButton = JButtonBuilder.builder()
+        .enabled(false)
+        .title("Add moderator")
+        .actionListener(button -> moderatorsTabActions.addModerator(addField, button, table))
+        .toolTip("Promotes a user to the moderators")
+        .build();
+
+    // This is a very fancy listener that will validate the entered requested new moderator
```

Comment:
@RoiEXLab not sure if you saw this bit of code, it's kinda interesting as the 'add moderator' button does not enable until the server has verified the user exists (this is done in background while user is typing). When testing this out I noticed the background thread was needed to not interrupt typing.